libcontainer/process_linux.go

```diff
@@ -254,9 +250,16 @@ func (p *setnsProcess) start() (retErr error) {
 			}
 		}
 	}
-	// Set final CPU affinity right after the process is moved into container's cgroup.
-	if err := p.setFinalCPUAffinity(); err != nil {
-		return err
+
+	if aff := p.config.CPUAffinity; aff == nil || aff.Final == nil {
+		if err := setAffinityAll(p.pid()); err != nil {
+			return err
+		}
+	} else {
+		// Set final CPU affinity right after the process is moved into container's cgroup.
+		if err := p.setFinalCPUAffinity(); err != nil {
+			return err
+		}
```

Comment:
Sorry, it's not obvious to me why if the cpu affinity fails, we should call the set finalCPUAffinity. Can you elaborate?